### PR TITLE
Fix ice_ids description

### LINKED ISSUE #1743: Incorrect documentation for ice_ids
The type IDs returned by ice_ids are sorted in alphabetical order (they are ASCII strings so ordinal comparison is fine).

However, the comments / doc-comments consistently claim they are sorted in "base-to-derived" order.

### PR DESCRIPTION
**Pull request by bernardnormier**
Fixes #1743.

### CODE REVIEW

File: csharp/src/Ice/Object.cs [hunk 1]
```diff
@@ -164,7 +163,7 @@ public static Task<OutputStream> iceD_ice_ids(Object obj, IceInternal.Incoming i
         /// Returns the Slice type ID of the most-derived interface supported by this object.
         /// </summary>
         /// <param name="current">The Current object for the invocation.</param>
-        /// <returns>The return value is always ::Ice::Object.</returns>
+        /// /// <returns>The return value is always ::Ice::Object.</returns>
```
Comment (InsertCreativityHere) [inline csharp/src/Ice/Object.cs:166]:
```suggestion
        /// <returns>The return value is always ::Ice::Object.</returns>
```

File: csharp/src/Ice/Proxy.cs [hunk 2]
```diff
@@ -248,8 +248,8 @@ public interface ObjectPrx
         /// Returns the Slice type IDs of the interfaces supported by the target object of this proxy.
         /// </summary>
         /// <param name="result">The asynchronous result object returned by <code>begin_ice_ids</code>.</param>
-        /// <returns>The Slice type IDs of the interfaces supported by the target object, in base-to-derived
-        /// order. The first element of the returned array is always ::Ice::Object.</returns>
+        /// <returns>The Slice type IDs of the interfaces supported by the target object, in in alphabetical order.
```
Comment (InsertCreativityHere) [inline csharp/src/Ice/Proxy.cs:251]:
```suggestion
        /// <returns>The Slice type IDs of the interfaces supported by the target object, in alphabetical order.
```

File: csharp/src/Ice/Proxy.cs [hunk 3]
```diff
@@ -1197,8 +1197,8 @@ public AsyncResult
         /// Returns the Slice type IDs of the interfaces supported by the target object of this proxy.
         /// </summary>
         /// <param name="result">The asynchronous result object returned by <code>begin_ice_ids</code>.</param>
-        /// <returns>The Slice type IDs of the interfaces supported by the target object, in base-to-derived
-        /// order. The first element of the returned array is always ::Ice::Object.</returns>
+        /// <returns>The Slice type IDs of the interfaces supported by the target object, n alphabetical order.
```
Comment (InsertCreativityHere) [inline csharp/src/Ice/Proxy.cs:1200]:
```suggestion
        /// <returns>The Slice type IDs of the interfaces supported by the target object, in alphabetical order.
```

File: java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java [hunk 4]
```diff
@@ -4,6 +4,12 @@
 
 package com.zeroc.Ice;
 
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.sql.Connection;
+
+import com.zeroc.IceGridGUI.LiveDeployment.Communicator;
+
```
Comment (InsertCreativityHere) [inline java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java:12]: What are all these imports for? O-O

**Review by InsertCreativityHere — APPROVED** [review]
Good to fix this!

**Review by pepone — APPROVED** [review]

File: java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java (hunk 4 above)
Comment (externl) [reply java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java:12]: I suspect they're a mistake caused by running the Java code formatter in VSCode

Comment (externl) [inline java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java:12]:
```suggestion
```

**Review by externl — APPROVED** [review]

**Pull request merged**